amazon/ion/ioncmodule.c

```diff
@@ -1428,7 +1427,7 @@ iERR ion_read_file_stream_handler(struct _ion_user_stream *pstream) {
 
     // safe-guarding the size variable to protect memcpy bounds
     if (size < 0  || size > IONC_STREAM_READ_BUFFER_SIZE) {
-        FAILWITH(IERR_READ_ERROR);
+        _FAILWITHMSG(IERR_READ_ERROR, "illegal bytes number has been read");
```

Comment:
Is this effectively an illegal state? If so, we don't really need the message. Could a user ever encounter this except in the case where there is a bug in the ion-python C extension?

(If we keep the message: "illegal bytes number has been read" -> "An illegal number of bytes have been read.")